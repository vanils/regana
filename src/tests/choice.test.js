
const regana = require('../regana.js');
const path = require('path');

describe('regana', () => {
  describe('#analyse', () => {
    describe('#choice', () => {
      test('analyse should work with if statement', () => {
        expect(regana.analyse(path.resolve(__dirname, './mocks/choice/if.js'))).toMatchSnapshot();
      });
    });
  });
});