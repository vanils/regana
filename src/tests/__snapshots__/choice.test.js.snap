// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regana #analyse #choice analyse should work with if statement 1`] = `
"[
  {
    \\"start\\": 0,
    \\"end\\": 15,
    \\"parentScope\\": null,
    \\"file\\": \\"File_0\\",
    \\"segments\\": {
      \\"IfStatement_1\\": {
        \\"uses\\": [],
        \\"start\\": 1,
        \\"end\\": 15,
        \\"id\\": \\"IfStatement_1\\"
      },
      \\"BooleanLiteral_2\\": {
        \\"uses\\": [],
        \\"start\\": 5,
        \\"end\\": 9,
        \\"id\\": \\"BooleanLiteral_2\\"
      },
      \\"BlockStatement_3\\": {
        \\"uses\\": [],
        \\"start\\": 11,
        \\"end\\": 15,
        \\"id\\": \\"BlockStatement_3\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {},
    \\"labels\\": {},
    \\"withSegments\\": []
  },
  {
    \\"start\\": 11,
    \\"end\\": 15,
    \\"parentScope\\": {
      \\"start\\": 0,
      \\"end\\": 15,
      \\"parentScope\\": null,
      \\"file\\": \\"File_0\\",
      \\"segments\\": {
        \\"IfStatement_1\\": {
          \\"uses\\": [],
          \\"start\\": 1,
          \\"end\\": 15,
          \\"id\\": \\"IfStatement_1\\"
        },
        \\"BooleanLiteral_2\\": {
          \\"uses\\": [],
          \\"start\\": 5,
          \\"end\\": 9,
          \\"id\\": \\"BooleanLiteral_2\\"
        },
        \\"BlockStatement_3\\": {
          \\"uses\\": [],
          \\"start\\": 11,
          \\"end\\": 15,
          \\"id\\": \\"BlockStatement_3\\"
        }
      },
      \\"exposes\\": {},
      \\"pointers\\": {},
      \\"labels\\": {},
      \\"withSegments\\": []
    },
    \\"file\\": \\"File_0\\",
    \\"segments\\": {},
    \\"exposes\\": {},
    \\"pointers\\": {},
    \\"labels\\": {},
    \\"withSegments\\": []
  }
]"
`;
